ls1intum/Artemis
# `Development`: Modernize chart visualization of active students and performance in exercises

### Checklist
#### General


#### Client
### Motivation and Context


This PR changes the line chart implementation of the _Active Students_ chart in each course card on the course management page and the _Performance in Exercises_ chart in the student statistics tab of every course from `Chart.js` to `ngx-charts`.

If you take a detailed look at both representations of _Active Students_, you might notice that the old one provides data labels displaying the absolute value for every point, which the new one doesn't. I'm aware of this difference, but wasn't able to find a possibility to implement this with `ngx-charts` so far.
### Description

The html templates as well as the underlying data processing logics in the components have been simplified and adapted in order to fit `ngx-charts` requirements.

### Steps for Testing

Prerequisites:
**Important:** before deploying this PR to a test server, take a look at the current implementation. This includes hovering over the line, adjusting screen size, clicking on nodes in the chart (_Performance in Exercises_ only) and clicking on parts of the legend (_Performance in Exercises_ only).

- 1 Instructor
- 1 Course with students (at least one) and exercises
- 1 student participating on this course

1. Log in to Artemis as instructor
2. Navigate to Course Administration
3. Inspect the new implementation of _Active Students_. Again, hover over the line and adjust the screen size. The chart should now be responsive to the screen width.
4. Log out and log in as student again
5. Click on the course card for the corresponding course and enter the statistics tab
6. Scroll down to the _Performance in Exercises_ chart and again, interact with it. Check whether the interaction is similar to the old implementation. The chart should be responsive to the screen width as well now.

### Review Progress





#### Code Review
#### Manual Tests

### Test Coverage





course-management-overview-statistics.component.ts: 66.67% | 90.32%

### Screenshots


**Old design of _Performance in Exercises_:**
![image](https://user-images.githubusercontent.com/80622272/141773334-51aa4926-c944-4c36-8ab4-b889d1aebd32.png)
**Old design of _Active Students_:**
![image](https://user-images.githubusercontent.com/80622272/141773071-c95e178d-bf26-4191-bbb0-3dd3b4f78a39.png)

**New design of _Performance in Exercises_:**
![image](https://user-images.githubusercontent.com/80622272/141951722-efc08dd7-2851-4c64-9585-b74f030e0e4c.png)
**New design of _Active Students_:**
![image](https://user-images.githubusercontent.com/80622272/141771544-df36eb51-6401-452e-9323-b2a5176dc909.png)
